torch_xla/csrc/runtime/pjrt_computation_client.cc

```diff
@@ -143,12 +144,17 @@ PjRtComputationClient::PjRtComputationClient() {
         "XLA_COORDINATOR_PORT", XlaCoordinator::kDefaultCoordinatorPort);
 
     // Use the XlaCoordinator as the distributed key-value store.
+    TF_VLOG(3) << "Getting StreamExecutorGpuClient for node_id="
+               << global_process_rank << ", num_nodes=" << global_world_size;
     coordinator_ = std::make_unique<XlaCoordinator>(
         global_process_rank, global_world_size, master_addr, port);
     std::shared_ptr<xla::DistributedRuntimeClient> distributed_client =
         coordinator_->GetClient();
-    auto allowed_devices =
-        std::make_optional<std::set<int>>(std::set{local_process_rank});
+    std::optional<std::set<int>> allowed_devices;
+    if (global_world_size > 1) {
+      allowed_devices =
+          std::make_optional<std::set<int>>(std::set{local_process_rank});
```

Comment:
nit: do you still need to `make_optional` here, or can you directly assign? e.g. `allowed_devices = std::set{local_process_rank}`